info/guide/guide-chapter-3.md

```diff
@@ -77,6 +77,60 @@ If the classes are meant to be used externally, and are not referenced inside th
 **Exception:**
 All variants of a `(private) val` logger should be placed at the beginning of the class (`(private) val log`, `LOG`, `logger`, etc.).
 
+#### <a name="r3.1.5"></a> 3.1.5 Order of declaration of top-level code structures
+Kotlin allows several top-level declaration types: classes, objects, interfaces, properties and functions.
+When declaring more than one class or zero classes (e.g. only functions), as per rule 2.2.1, you should document the whole file in the header KDoc.
+When declaring top-level structures, keep the following order:
+1. Top-level public constants (`const val`)
+2. Top-level public properties
+   1.1 `val`
+   1.2 `lateinit var`
+   1.3 `var`
+3. Public classes
+4. Public extension functions
+5. Public functions
+6. Private constants and properties, following same order as in 1 and 2
+7. Private classes
+8. Private extension functions
+9. Private functions
+
+**Note**:
+Extension functions, that use the class or interface declared in the same file (accept parameters of this type or return this type),
```

Comment:
please create a feature to check that no one creates extension function in the same file with the class

Reply:
#731